### PR TITLE
r_d7: R-P-S game

### PR DESCRIPTION
**Pull request by ShekhaninKyryl**
МІНІ-ГРА в консолі: Камінь Ножиці Папір

```
1. Користувач вводить кількість раундів гри N;
2. N раундів користувач вводить свій вибір: r (rock), p (paper), s (scissor), комп’ютер генерує свій вибір випадковим чином (або за допомогою іншого алгоритму на ваш вибір)
3. по закінченню N раундів в користувача наявна статистика по виграним/програним/нічийним дуелям та фінальний результат за сумою всіх раундів: більше виграних дуелей, програних або ж нічия
4. Повторити з кроку 1., поки юзер не введе -1 як кількість раундів для закінчення гри;
```

### CODE REVIEW

File: r_d7/GameRules.cpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+char convertVariantToSymbol(Variant value)
+{
+	switch (value)
+	{
+	case Variant::Rock: return 'r';
+	case Variant::Paper: return 'p';
+	case Variant::Scissors: return 's';
+	default: return 'u';
+	}
+	return 0;
```
Comment (bohdanpc) [inline r_d7/GameRules.cpp]: not needed, as long as u have default

File: r_d7/GameRules.cpp [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+}
+
+void showPossibleVariant()
+{
+	std::cout << '\t' << "r - rock" << std::endl;
+	std::cout << '\t' << "p - paper" << std::endl;
+	std::cout << '\t' << "s - scissors" << std::endl;
+
+	return;
```
Comment (bohdanpc) [inline r_d7/GameRules.cpp]: redundant

File: r_d7/GameRules.cpp [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+Variant convertToVariant(char symbol)
+{
+
+	if (symbol == 'r' || symbol == 'R') return Variant::Rock;
+	if (symbol == 'p' || symbol == 'P') return Variant::Paper;
+	if (symbol == 's' || symbol == 's') return Variant::Scissors;
+	return Variant();
+}
+
+void showStatistics(int winCount, int drawCount, int loosCount)
```
Comment (bohdanpc) [inline r_d7/GameRules.cpp]: loseCount

File: r_d7/GameRules.cpp [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+	}
+	else {
+		std::cout << "!!!smth wrong!!!" << std::endl;
+	}
+
+	std::cout << "-------------------------" << std::endl;
+	return;
+}
+
+RoundResult gameRaund(int& raundNumber, bool& isExit)
```
Comment (bohdanpc) [inline r_d7/GameRules.cpp]: gameRound

File: r_d7/GameRules.cpp [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+{
+	// generate choose form -1 to 1
+	const Variant randomChoose = convertToVariant(std::rand() % VARIANT_LENGTH);
+	Variant userChoose;
```
Comment (bohdanpc) [inline r_d7/GameRules.cpp]: choice

File: r_d7/GameRules.cpp [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+		return sum == 2 ? -1 : sum;
+	}
+	//1-P	1	2	3
+	case Variant::Paper: {
+		return sum == 1 ? -1 : sum % 2;
+	}
+	//2-S	2	3	4
+	case Variant::Scissors: {
+		return sum == 2 ? 1 : sum - 4;
+	}
```
Comment (bohdanpc) [inline r_d7/GameRules.cpp:159]: So, relying on particular order of the values in enum class Variant?
Changing places of the values inside of the enum will break this logic? Doesn't look too robust.
Simple conditions would do the trick

**Comment by ShekhaninKyryl** [comment]
Thank you for comments. I fixed naming issue and implemented checkWinnerByConditions. Please rereview pr

**Pull request merged**